packages/protocol/contracts/L2/DelegateOwner.sol

```diff
@@ -69,27 +77,35 @@ contract DelegateOwner is EssentialContract, IMessageInvocable {
         payable
         onlyFromNamed(LibStrings.B_BRIDGE)
     {
-        (uint64 txId, address target, bytes memory txdata) =
-            abi.decode(_data, (uint64, address, bytes));
-
-        if (txId != nextTxId) revert DO_INVALID_TX_ID();
-
         IBridge.Context memory ctx = IBridge(msg.sender).context();
         if (ctx.srcChainId != l1ChainId || ctx.from != realOwner) {
             revert DO_PERMISSION_DENIED();
         }
-        nextTxId++;
-        // Sending ether along with the function call. Although this is sending Ether from this
-        // contract back to itself, txData's function can now be payable.
-        (bool success,) = target.call{ value: msg.value }(txdata);
-        if (!success) revert DO_TX_REVERTED();
+        _invokeCall(_data, true);
+    }
 
-        emit TransactionExecuted(txId, target, bytes4(txdata));
+    /// @notice Dryruns a message invocation but always revert.
+    /// If this tx is reverted with DO_TRY_RUN_SUCCEEDED, the try run is successful.
+    function dryrunMessageInvocation(bytes calldata _data) external payable {
+        _invokeCall(_data, false);
+        revert DO_DRYRUN_SUCCEEDED();
```

Comment:
Is it just to test the call, to help with execution evaluation, right ?

Reply:
It actually runs all the inner transactions and if they don't revert, a final `DO_DRYRUN_SUCCEEDED` revert will indicate all inner transactions are successful. But this tx burns ETH.